test/test_torch.py

```diff
@@ -10465,6 +10465,61 @@ def run_test(tensor_dims, some):
         for tensor_dims, some in product(tensor_dims_list, [True, False]):
             run_test(tensor_dims, some)
 
+    @dtypes(torch.float, torch.double)
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_quantile(self, device, dtype):
+
+        # Generate some random test cases
+        a_sizes = [tuple(np.random.randint(2, 10, size=i)) for i in range(1, 4)]
+        q_values = [tuple(np.random.rand(i)) for i in range(0, 4)]
+
+        # Add corner cases
+        a_sizes.extend([(1,), (1, 1), (1, 2, 1)])
+        q_values.extend([0.5, (0., 1.), np.random.rand(10)])
+
+        # Enumerate all input combinations
+        for test_case in product(a_sizes, q_values, [True, False]):
+            a = torch.randn(test_case[0], dtype=dtype, device=device)
+            q = torch.tensor(test_case[1], dtype=dtype, device=device)
+            keepdim = test_case[2]
+            # Compute quantile along every dimension and flattened tensor
+            for dim in [None] + list(range(a.ndim)):
+                result = torch.quantile(a, q, dim=dim, keepdim=keepdim).cpu()
+                expected = np.quantile(a.cpu().numpy(), q.cpu().numpy(), axis=dim, keepdims=keepdim)
+                expected = torch.from_numpy(np.array(expected)).type(result.type())
+                self.assertTrue(torch.allclose(result, expected, rtol=1e03, atol=1e06))
```

Comment:
1e-03 and 1e-06? 

Was there a problem with the defaults?

Reply:
I've been struggling to find good numbers for atol and rtol because NumPy promotes float to double and hence has better accuracy. For double it works fine, but for floats I keep failing tests even though the numbers are pretty close to about 5 digits.

Reply:
Yes, because NumPy promotes float to double I'm struggling to find numbers that work for the float case.